mqtt/mqtt-broker/src/auth.rs

```diff
@@ -0,0 +1,85 @@
+#![allow(dead_code)] // TODO @dmolokanov remove when implemented
+use std::fmt;
+
+use failure::Fail;
+
+use crate::Error;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum AuthId {
+    Any,
+    Value(Identity),
+}
+
+pub type Identity = String;
```

Comment:
Should you wrap this instead? 

https://github.com/rust-unofficial/patterns/blob/master/patterns/newtype.md

Reply:
It's not clear for now what type is going to be for Indentity 
just `struct Identity(String)`
or `enum Identity { Local(String), IotHub(String) }` 
so I decided to leave it as an alias for now